internal/testrunner/runners/system/servicedeployer/terraform_env.go

```diff
@@ -25,6 +28,9 @@ func (tsd TerraformServiceDeployer) buildTerraformExecutorEnvironment(ctxt Servi
 	vars[tfTestRunID] = ctxt.Test.RunID
 	vars[tfDir] = tsd.definitionsDir
 
+	os.MkdirAll(outputDir+ctxt.Test.RunID, os.ModePerm)
```

Comment:
Can you store this directory path within the `ServiceContext`. I see this path being recreated in a multiple places and it would more clear and less prone to future errors if it were constructed and stored in just one place.

This is missing an error check.

Why `os.ModePerm`? Can we use `0o755`?